src/main/java/org/candlepin/subscriptions/product/OfferingSyncController.java

```diff
@@ -171,12 +178,35 @@ private SyncResult syncOffering(Offering newState, Optional<Offering> persistedO
       }
     }
 
-    // Existing capacities might need updated if certain parts of the offering was changed.
-    capacityReconciliationController.enqueueReconcileCapacityForOffering(newState.getSku());
+    // Existing capacities need to be updated if certain parts of the offering changed.
+    if (isCapacityImpacted) {
```

Comment:
Should this enqueue functionality occur before saveAndFlush in offeringRepo? This doesn't seem incorrec, as the save offering part is async and will also rollback if the enqueue transaction fails. If this implementation feels appropriate, I am fine.

Reply:
Per [spring-kafka docs](https://docs.spring.io/spring-kafka/reference/kafka/transactions.html#container-transaction-manager):

> When the method exits, the database transaction will commit followed by the Kafka transaction.

In other words, the messages won't be sent if the DB transaction fails. If this was triggered as part of processing a task message, the message will be reprocessed. If it was triggered by an API, the API should show the error. In either case, we'd either manually (via API) or automatically (via kafka message) retry.

Also, it's been this way for a while, I just made the message send conditional.

Reply:
yup  :100: . I don't see any issue as I mentioned but, just wanted your thoughts on it.